src/main/java/com/minecolonies/core/entity/ai/workers/crafting/EntityAIConcreteMixer.java

```diff
@@ -42,136 +44,156 @@ public class EntityAIConcreteMixer extends AbstractEntityAICrafting<JobConcreteM
     public EntityAIConcreteMixer(@NotNull final JobConcreteMixer job)
     {
         super(job);
+        super.registerTargets(
+          new AITarget(CONCRETE_MIXER_PLACING, this::placePowder, TICKS_SECOND),
+          new AITarget(CONCRETE_MIXER_HARVESTING, this::harvestConcrete, TICKS_SECOND)
+        );
     }
 
-    @Override
```

Comment:
the reogs make this class again very difficult to review

Reply:
It still may look a little messy, but there's actually no reorg going on. The methods just changed that much.

Reply:
There are 3 new functions up top, `placePowder`, `harvestConcrete` and `getSlotWithPowder`.
The `decide` task was not changed, only it's commented out code removed.
The `craft` task was changed quite a bit, because it's logic was split up into the place and harvest methods up top.
One new method at the bottom, `performMixingWork`.